.github/workflows/vercel-test-deploy.yml

```diff
@@ -23,3 +27,21 @@ jobs:
         cache: 'npm'
     - run: npm ci
     - run: npm test
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: test
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Vercel CLI
+        run: npm install --global vercel@latest
+      - name: Pull Vercel Environment Information
+        run: vercel pull --yes --environment=preview --token=${{ secrets.VERCEL_TOKEN }}
```

Comment:
I don't think this will work, since depending on the branch we need to pull a different environment variable.

any `/release/xxx` or main branch should pull production
any `feature/xxx` branch should pull preview
any other branch should pull development